[docs]: add configuration reference

0x009922 opened the pull request:
Related issue: #392 

### Checklist

- [ ] Adapt old "Configuration and Management" articles to the new reference
- [x] Agree on the new configuration endpoints
- [ ] Document configuration ISIs
- [x] Assign ENVs to parameters

WRRicht3r commented on src/api/config/index.md:
````diff
@@ -0,0 +1,124 @@
+api_address = "localhost:8080"
+```
+
+The path to the configuration file might be overwritten either with `--config` CLI argument, or with `IROHA_CONFIG` environment variable: 
````
```suggestion
The path to the configuration file can be overwritten either with `--config` CLI argument, or with `IROHA_CONFIG` environment variable: 
```

WRRicht3r commented on src/api/config/index.md:
```diff
@@ -0,0 +1,124 @@
+
+:::
+
+Please refer to each field's documentation for specific ENV names.
```
```suggestion
Please refer to the documentation from each field for specific ENV names.
```

0x009922 replied:
"From" each field, of "for" each field?

outoftardis replied:
for

WRRicht3r commented on src/api/config/index.md:
```diff
@@ -0,0 +1,124 @@
+
+## TOML Format
+
+Iroha uses [TOML (Tom's Obvious Minimal Language)](https://toml.io/) format for its configuration. Please refer to its documentation if you need assistance about how it works.
```
```suggestion
Iroha uses the [TOML (Tom's Obvious Minimal Language)](https://toml.io/) format for its configuration. Please refer to its documentation if you need assistance with how it works.
```

WRRicht3r commented on src/api/config/kura-params.md:
```diff
@@ -0,0 +1,34 @@
+
+::: tip Dev Note
+
+Validation of this parameter is kind of delayed. What if error appears very
```
```suggestion
Validation of this parameter is delayed. What if an error appears very
```

WRRicht3r commented on src/api/config/sumeragi-params.md:
```diff
@@ -0,0 +1,107 @@
+
+Hey, is sumeragi.trusted_peers actually required to be set in the config?
+
+I guess that's the only way to specify the relevant topology and let the nodes know what addresses to communicate with as well as what public keys to use to verify respective node signatures
```
No guessing, either yes or no please.

0x009922 replied:
This text is not for the users... it is for us to discuss.

WRRicht3r commented on src/api/config/telemetry-params.md:
```diff
@@ -0,0 +1,47 @@
+
+## `telemetry.substrate`
+
+Enable Substrate active outbound telemetry.
```
```suggestion
Enables Substrate active outbound telemetry.
```

WRRicht3r commented on src/api/config/telemetry-params.md:
```diff
@@ -0,0 +1,47 @@
+
+## `telemetry.file_output`
+
+Enable file-based active outbound telemetry.
```
```suggestion
Enables file-based active outbound telemetry.
```

WRRicht3r commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,61 @@
+
+::: info
+
+This is different from [Telemetry](telemetry-params) section. This parameter is
```
```suggestion
This section differs from the [Telemetry](telemetry-params) section. This parameter is
```

WRRicht3r commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,61 @@
+::: info
+
+This is different from [Telemetry](telemetry-params) section. This parameter is
+about "passive" telemetry, requested by external actors. That section is
```
```suggestion
regarding "passive" telemetry requested by external actors, while that section is
```

WRRicht3r commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,61 @@
+- **Type:** String or Number, [Byte Size](glossary#type-byte-size)
+- **Default:** $2^{15} = 32\ 768$
+
+Maximum number of bytes in raw transaction. Used to prevent from DOS
```
```suggestion
The maximum number of bytes in a raw transaction. Used to prevent DOS
```

WRRicht3r commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,61 @@
+- **Type:** String or Number, [Byte Size](glossary#type-byte-size)
+- **Default:** $2^{12} \cdot 4\ 000 = 16\ 384\ 000$
+
+Maximum number of bytes in raw message. Used to prevent from DOS attacks.
```
```suggestion
The maximum number of bytes in a raw message. Used to prevent DOS attacks.
```

WRRicht3r commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,61 @@
+- **Type:** String or Number, [Duration](glossary#type-duration)
+- **Default:** 30 seconds
+
+Time query can remain in the store if unaccessed
```
```suggestion
The time a query can remain in the store if unaccessed
```

WRRicht3r commented on src/api/config/wsv-params.md:
```diff
@@ -0,0 +1,75 @@
+
+## `wsv.ident_length_limits`
+
+Limits for the number of characters in identifiers that can be stored in
```
```suggestion
Limits to the number of characters in identifiers that can be stored in
```

0x009922 replied:
Not sure

6r1d commented on src/api/config/base-params.md:
```diff
@@ -0,0 +1,23 @@
+# Base Parameters
```
Are there any requirements for the key?

* Could I introduce a new peer with random keys (given a proper algorithm that generates them and that they have a proper format)?
* Could I randomly change keys for a peer at some moment, restart it and expect it to work?

0x009922 replied:
1. Well, each peer might have any keys you want it to use. I lack understanding of the bigger picture and how keys should be approached in real-world scenarios.
2. Not sure, don't think so.

6r1d commented on src/api/config/block-sync-params.md:
```diff
@@ -0,0 +1,27 @@
+# Block Sync Parameters
+
+Explain module
```
Honestly, I haven't heard about that part before, so any explanation would be helpful. I wonder who can help?

0x009922 commented on src/api/api.md:
```diff
@@ -0,0 +1,20 @@
+This page contains a copy of `api_spec.md` from
+`hyperledger/iroha#iroha2-dev`. You can read the most up-to-date version of
+it on
+[GitHub](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/references/api_spec.md).
```
Btw, I think we should remove `api_spec.md` from Iroha repo and put it in the docs. The same approach I am implementing now with the config.

yamkovoy replied:
I 100% agree with @0x009922

0x009922 replied:
Created an issue: https://github.com/hyperledger/iroha-2-docs/issues/414

0x009922 commented on src/api/config/base-params.md:
```diff
@@ -0,0 +1,23 @@
+- **Type:** String, [Multi-hash](glossary#type-multi-hash)
+- **Required**
+
+Public key of this peer
```
1. I can hardly answer to it
2. Yes, `kagami crypto` might generate a usable key pair

0x009922 replied:
I don't think that we should mention keys generation and Kagami in the configuration reference. There are many other places in the docs which mention keys, and explaining the same things again and again might be ridiculous and inconsistent. I would like to have a single "Keys 101" section which we could reference from everywhere.

outoftardis commented on src/api/config/genesis-params.md:
```diff
@@ -0,0 +1,50 @@
+# Genesis Parameters
+
+Explain the purpose of this block. Maybe explain both keys in a single
+section?
```
this is a comment, not the content, so can you please have this (and similar instances) commented out?

yamkovoy replied:
Great suggestion!

0x009922 replied:
Used TODOs

yamkovoy commented on src/api/config/block-sync-params.md:
```diff
@@ -0,0 +1,27 @@
+## `block_sync.gossip_period`
+
+- **Type:** String or Number, [Duration](glossary#type-duration)
+- **Default:** 10 seconds
```
is it seconds <seconds> or <secs> as in the example below? What will actually be accepted by the `gossip_period` parameter?

0x009922 replied:
`gossip_period` accepts a **String** or a **Number**, and interprets it as a **Duration**. Duration is explained at the link.

For Duration types, I specify their "Default" values as a human readable string like "30 seconds" or "5 minutes".

In the examples, I specify any valid value. Here is a question: maybe, we should change example to show the default value for each config parameter?

yamkovoy replied:
My last comment doesn't have a part of the message for some reason. I was actually interested in the terminology.
In the "Default" you have "10 seconds", while in the example you have "5 secs". 
So, the question is: does a peer have to write out "seconds" or "secs", or just a plain number in the "gossip_period" field? Juding by your reply, it seems like peers can input minutes as well.
For instance, if I want the interval to be 3 minutes and 22 seconds, should I input "3 mins, 22 secs" or "202 secs", or any other variation?
I'd like to make it clear for the readers

0x009922 replied:
I see, and it is why I put the link `[Duration](glossary#type-duration)`. That section explains all the possible inputs for durations.

Should it be pointed out more explicitly in some way?

yamkovoy commented on src/api/config/genesis-params.md:
```diff
@@ -0,0 +1,50 @@
+
+
+
+The warning will be printed if the
```
What exactly does the warning say? What happens after the warning is printed?

0x009922 replied:
Let's come up with a line!

I want to remind that this configuration reference describes the configuration that doesn't exist yet. We are at the design stage, and we need to find the best ways to present configuration system to the users. It means that we might change the naming/structure/phrasing of anything if we decide it will be more convenient to users.

As for this topic, actually Iroha should fail to start with `--submit-genesis` arg, but without `genesis.private_key` config parameter. However, if there is no `--submit-genesis` arg, but `genesis.private_key` is provided, what Iroha should do? Just print a warning and continue execution, or fail?

@Mingela, your input might be valuable.

Mingela replied:
Since the private key is needed only when a flag is specified it feels like the values should be combined somehow, i.e. `--submit-genesis --private-key <key>`. I might have encountered flag dependencies where one couldn't be used without other but cannot find an example. So following my suggestion just `--private-key <key>` must prompt an error and usage manual (like `usage: iroha [--submit-genesis --private-key <key>] .. etc`).
Another idea came to my mind, what about removing `--submit-genesis` at all and treat presence of `--private-key <key>` implicitly as it's intended to submit the genesis block?

Mingela replied:
`private-key` was used for simplicity there, we should define a simpler name for `genesis.private_key` in that case, maybe `genesis-key`?

0x009922 replied:
There are two parameters:

- `genesis.public_key`: used by all peers
- `genesis.private_key`: used only by the peer which submits the genesis

Not sure how `genesis-key` (or `genesis.key`) might be used in this context.

The downside of `iroha --submit-genesis --private-key <key>`: it is not good to provide sensitive information through CLI. Although it might be workarounded with ENV: `--private-key $IROHA_PRIV_KEY`.

Anyway, the solution of omitting `--submit-genesis` seems fine. The main downside of it as I see is that it might not be that obvious which peer submits the genesis, and the user must know the details of how configuration works.

Still, if we want to choose the path of the highest reliability for the end user, I think the current approach is the best. You clearly identify your intention with `--submit-genesis` flag. Iroha will clearly fail if you forgot to provide `genesis.private_key` (or `GENESIS_PRIVATE_KEY` var (naming is up to discussion)). In this case, the question becomes: should Iroha **fail or warn** if the key is provided, but the intention was not given?

Mingela replied:
Either is fine. Seems all options around the `--submit-genesis` have tradeoffs. I don't like the ability to have unused parameters specified, though a warning would be reasonable in that case. Let's hear opinions of others, @mversic @BAStos525

BAStos525 replied:
We definitely should not provide `genesis.private.key` as shell parameter and it has to be hided somehow, e.g. env var. Currently, we have [this block](https://github.com/hyperledger/iroha/blob/df343f3390829121d193fa85fd10d540ccbe41d9/configs/peer/config.json#L50) in peer `config.json` file, and we pass these keys in plain text. Should it be hided somehow? Moreover, we don't separate this config file between peers, so it's the same for all peers. We manage genesis submitting only by `--submit-genesis` for `peer-0` selection by script. 
So, there are two ways:
  1. Should we pass genesis keys through `config.json`? Should it be hided and private key should be specified only for genesis peer-0 (different configs for peers then). And if it's necessary to use `--submit-genesis` flag for `peer-0` if keys are already specified in config?
  2. Remove `--submit-genesis` flag and this `genesis.private.key` field in config and apply genesis private key through ENV var or any kind of secret.

The second item sounds a kinda better since it may be easier to hide a sensitive data.
As for _should iroha if the keys are provided, but the intention was not given_, I think it depends on the selected approach how to provide the private key. If through config and flag, so it means to have a more deployment routine to separate peers configs e.t.c. to avoid iroha fail. So, here warning sounds more reasonable. In the second option we could try to fail it in order do not spread sensitive data across peers where and if it's not necessary.

0x009922 replied:
We had a discussion with the team and dove deeper into the topic of genesis. We agreed that the whole genesis approach as of now is somewhat weird and needs thoughtful revision. It will be done later (idk when, hopefully soon). As for configuration, I will preserve the current behavior without significant changes.

That is:

- `genesis.public_key` should be provided for all peers. Through config file **or** ENV var.
- `genesis.private_key` should be provided for the peer with `--submit-genesis` CLI flag. It might be set through config file **or** ENV var.
- if `genesis.private_key` is provided, but `--submit-genesis` is not, then Iroha prints a warning, but continues execution.
- if `--submig-genesis` is provided, but `genesis.private_key` is not, then Iroha fails with FATAL error.

outoftardis commented on src/api/config/glossary.md:
````diff
@@ -0,0 +1,131 @@
+address = "localhost:8000"
+```
+
+## Type - Metadata Limits
````
```suggestion
## Type: Metadata Limits
```

outoftardis commented on src/api/config/glossary.md:
````diff
@@ -0,0 +1,131 @@
+private_key = { digest = "ed25519", payload = "" }
+```
+
+## Type - Socket Address
````
```suggestion
## Type: Socket Address
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+TODO: [article explains kb, kB, KiB difference](https://web.archive.org/web/20150324153922/https://pacoup.com/2009/05/26/kb-kb-kib-whats-up-with-that/).
+Consider it for the format.
+
+## Type - Private Key
```
```suggestion
## Type: Private Key
```

will there be any explanation?

0x009922 replied:
Yes, let me put a TODO.

outoftardis commented on src/api/config/glossary.md:
````diff
@@ -0,0 +1,131 @@
+value = 512 # 512 bytes
+```
+
+String:
````
```suggestion
Byte size as a string:
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+- As a Number, which will be considered an amount in bytes
+- As a String, which will be parsed as a human_readable byte size string
+
+Number:
```
```suggestion
Byte size as a number:
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+Byte size values might be set in two ways:
+
+- As a Number, which will be considered an amount in bytes
+- As a String, which will be parsed as a human_readable byte size string
```
```suggestion
- As a String, which will be parsed as a human-readable string
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+
+Describe what the hell this is
+
+## Type - Byte Size
```
```suggestion
## Type: Byte Size
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+
+## Type - Multi-hash
+
+Describe what the hell this is
```
```suggestion
TODO Describe what the hell this is
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+[`humantime` crate](https://docs.rs/humantime/latest/humantime/fn.parse_duration.html)?
+It is an implementation detail.
+
+## Type - Multi-hash
```
```suggestion
## Type: Multi-hash
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+Duration might be specified in two ways:
+
+- As a Number, which will be considered an amount in milliseconds
+- As a String, which will be parsed as a human_readable duration string
```
```suggestion
- As a String, which will be parsed as a human-readable string
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+
+TODO Explain the limitations of different numeric types, like `u8` and `u64`.
+
+## Type - Duration
```
```suggestion
## Type: Duration
```

outoftardis commented on src/api/config/glossary.md:
```diff
@@ -0,0 +1,131 @@
+# Glossary
+
+This page contains the definitions and brief explanations of some types and topics shared across
```
```suggestion
This page contains the definitions and brief explanations of data types and topics shared across
```

outoftardis commented on src/api/config/kura-params.md:
```diff
@@ -0,0 +1,30 @@
+
+## `kura.blocks_per_storage_file`
+
+- **Type:** non zero u64
```
```suggestion
- **Type:** non-zero u64
```

outoftardis commented on src/api/config/kura-params.md:
```diff
@@ -0,0 +1,30 @@
+
+## `kura.init_mode`
+
+- **Type:** String
+- **Possible Values:** `strict` or `fast`
+- **Default:** `strict`
+
+`strict` - Strict validation of all blocks. `fast` - Fast initialization
+with basic checks.
```
```suggestion
- **Type:** String
- **Possible Values:** `strict` (strict validation of all blocks) or `fast` (fast initialization with basic checks)
- **Default:** `strict`
```

```suggestion
- **Type:** String
- **Possible Values:**
  - `strict`: strict validation of all blocks
  - `fast`: fast initialization with basic checks
- **Default:** `strict`
```

outoftardis commented on src/api/config/logger-params.md:
```diff
@@ -0,0 +1,56 @@
+
+- **ENV:** `LOG`, `LOG_LEVEL`, `IROHA_LOG_LEVEL`
+- **Type:** String
+- **Possible Values:** `"TRACE"`, `"DEBUG"`, `"INFO"`, `"WARN"`, `"ERROR"`,
```
will you explain what each of the value mean?

0x009922 replied:
Yup, will do (or put a todo, at least)

outoftardis commented on src/api/config/logger-params.md:
```diff
@@ -0,0 +1,56 @@
+
+::: warning
+
+This will produce a warning:
```
```suggestion
Setting `colorize_output` to `true` will produce a warning:
```

outoftardis commented on src/api/config/network-params.md:
```diff
@@ -0,0 +1,11 @@
+- **Type:** String, [Socket-Address](glossary#type-socket-address)
+- **Required**
+
+address for p2p communication for consensus and block synchronization
```
```suggestion
Address for p2p communication for consensus and block synchronization
```

outoftardis commented on src/api/config/queue-params.md:
```diff
@@ -0,0 +1,34 @@
+- **Default:** $2^{16} = 65\ 536$
+
+The upper limit of the number of transactions waiting in the queue for
+single user. Use this option to apply throttling.
```
```suggestion
a single user. Use this option to apply throttling.
```

outoftardis commented on src/api/config/snapshot-params.md:
```diff
@@ -0,0 +1,41 @@
+- **Type:** String or Number, [Duration](glossary#type-duration)
+- **Default:** 1 minute
+
+Frequency of making snapshot
```
```suggestion
Frequency of making snapshots
```

outoftardis commented on src/api/config/block-sync-params.md:
```diff
@@ -0,0 +1,27 @@
+
+## `block_sync.block_batch_amount`
+
+- **Type:** Number
+- **Default:** $4$
+
+The number of blocks that can be sent in one message.
```
I wonder why the explanation of a config parameter goes after its specifications (type, values, default value, etc.). Why not change this order?

0x009922 replied:
I have no objections, and you probably understand better how it would be more comprehensible. Should I change the order everywhere?

outoftardis commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,60 @@
+- **Type:** String, [Socket-Address](glossary#type-socket-address)
+- **Required**
+
+Address for client API.
```
```suggestion
Address of the client API.
```

outoftardis commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,60 @@
+- **Type:** String, [Socket-Address](glossary#type-socket-address)
+- **Optional**
+
+address for reporting internal status and metrics for administration.
```
```suggestion
Address for reporting internal status and metrics for administration.
```

outoftardis commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,60 @@
+
+::: info
+
+This section differs from the[Telemetry](telemetry-params) section. This
```
```suggestion
This section differs from the [Telemetry](telemetry-params) section. This
```

outoftardis commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,60 @@
+- **Type:** Number
+- **Default:** $10$
+
+How many query results are returned in one batch
```
```suggestion
The number of query results returned in one batch
```

outoftardis commented on src/api/config/telemetry-params.md:
```diff
@@ -0,0 +1,47 @@
+
+**Fields:**
+
+- **`file`:** The filepath that to write dev-telemetry to
```
```suggestion
- **`file`:** The filepath to write dev-telemetry to
```

outoftardis commented on src/api/config/telemetry-params.md:
```diff
@@ -0,0 +1,47 @@
+- **`name`:** The node's name to be seen on the telemetry
+  - **Type:** String
+  - **Required**
+- **`url`:** The url of the telemetry
```
in other places you use "address", not "url". Shouldn't it be standartized?

0x009922 replied:
There is a difference between URL and address. Here it is exactly an URL. For example, for other `address`es, users might specify `localhost:8080` (host + port), but for URLs, users should specify the full path like `http://localhost:8081/telemetry` (protocol + host + port + path etc).

outoftardis replied:
ok, got it! thanks

outoftardis commented on src/api/config/snapshot-params.md:
```diff
@@ -0,0 +1,61 @@
+- **Type:** String
+- **Possible Values:**
+  - **`normal`:** Iroha creates snapshots with a period specified by `snapshot.create_every`. On startup, Iroha reads an existing snapshot (if any) and verifies that it is up-to-date with the blocks storage.
+  - **`read-only`:** Like `normal`, but Iroha doesn't create any snapshots.
```
```suggestion
  - **`read-only`:** Similar to `normal` but Iroha doesn't create any snapshots.
```

outoftardis commented on src/api/config/index.md:
```diff
@@ -0,0 +1,133 @@
+:::
+
+When running `iroha` from the command line, Iroha will try to resolve a
+config file named `iroha.toml` inside `cwd`. Then it tries to read a set of
```
```suggestion
config file named `iroha.toml` inside the current working directory. Then it tries to read a set of
```

outoftardis commented on src/api/config/index.md:
```diff
@@ -0,0 +1,133 @@
+
+When running `iroha` from the command line, Iroha will try to resolve a
+config file named `iroha.toml` inside `cwd`. Then it tries to read a set of
+Environment Variables, associated with configuration parameters. The most
```
```suggestion
Environment Variables associated with configuration parameters. The most
```

0x009922 commented on src/api/config/genesis-params.md:
```diff
@@ -0,0 +1,50 @@
+
+The warning will be printed if
+[`genesis.private_key`](#genesis-private-key) and
+[`--submit-genesis`](../cli#submit-genesis) are used without each other.
```
`--submit-genesis` is not a command, but a CLI (command-line interface) flag. It might be set or not.

0x009922 commented on src/api/config/logger-params.md:
\*destination, it is a typo, will fix

0x009922 commented on src/api/torii-endpoints.md:
```diff
@@ -152,6 +152,14 @@ Learn [how to use metrics](/guide/advanced/metrics).
+
+::: tip Lazily Evaluated Pagination
+
+TODO: explain how it works. Explain that this behaviour is configured with [`torii.query_results_per_fetch`](/api/config/torii-params#torii-query-results-per-fetch) and [`torii.query_idle_time`](/api/config/torii-params#torii-query-idle-time).
```
@mversic please help

mversic replied:
@arjentix has been making changes recently. I think he can give more correct overview

Arjentix replied:
Hey, sorry for the delay.

So basically after https://github.com/hyperledger/iroha/pull/3958 users should send either `SignedQuery` with `Pagination` and `Sorting` or just `ForwardCursor`.

`SignedQuery` -- an Iroha query signed by a user keypair
`Pagination` -- struct with two numbers: limit of indexing and start of indexing. If both are zeros, then pagination is not applied
`Sorting` -- currently we support sorting only by a metadata key of an asset metadata. So that user can place `Name` (string with some contrains on its content) inside and then all output assets will be sorted by a value of the specified key if there is some. `Sorting` can contain nothing as well, in that case it won't be applied.
`ForwardCursor` -- at first you don't have it. But if the output of your query is a big array, then you'll get a cursor from Iroha with a **part** of the answer. To get the other parts you should provide this cursor to Iroha in a new request. Repeat this step until cursor is depleted (you'll know that by getting empty cursor or error).

I think you can use this info to provide a good documentation.

0x009922 replied:
Moved: https://github.com/hyperledger/iroha-2-docs/issues/426

outoftardis commented on src/api/config/sumeragi-params.md:
```diff
@@ -0,0 +1,195 @@
+- **Type:** String or Number, [Duration](glossary#type-duration)
+- **Default:** 10 seconds
+
+The time interval between requests to peers for the most recent block.
```
`requests to peers for the most recent block` does this mean requests *about* the most recent block?

0x009922 replied:
Yes. Request the most recent block from peers.

outoftardis replied:
then maybe

```suggestion
The time interval between sending requests to peers about the most recent block.
```

outoftardis commented on src/api/config/sumeragi-params.md:
```diff
@@ -0,0 +1,195 @@
+- **Type:** Number
+- **Default:** $4$
+
+The amount of blocks that can be sent in a single synchronization message.
```
```suggestion
The number of blocks that can be sent in a single synchronization message.
```

outoftardis commented on src/api/config/sumeragi-params.md:
```diff
@@ -0,0 +1,195 @@
+- **Type:** Number
+- **Default:** $500$
+
+Max number of transactions in a gossip batch message. Smaller size leads to longer time to synchronise, but useful if
```
```suggestion
Max number of transactions in a gossip batch message. Smaller size leads to longer time to synchronise, but can be useful if
```

outoftardis commented on src/api/config/sumeragi-params.md:
```diff
@@ -0,0 +1,195 @@
+
+::: warning
+
+More frequent gossiping shortens the time to sync, but can overload the network.
```
```suggestion
More frequent gossiping shortens the time to sync but can overload the network.
```

outoftardis commented on src/api/config/sumeragi-params.md:
```diff
@@ -0,0 +1,195 @@
+
+The upper limit of the number of transactions per block. If there is enough transactions in the [Queue](queue-params),
+the block is created immediately. Otherwise, the block is created when [`sumeragi.block_time`](#sumeragi-block-time) is
+elapsed since the round start[^1].
```
```suggestion
elapsed since the start of the round[^1].
```

outoftardis commented on src/api/config/sumeragi-params.md:
````diff
@@ -0,0 +1,195 @@
+```
+
+[^1]:
+    The round start happens on peers rotation, when the leader is elected. Generally it happens after the previous block
+    is committed. See [Consensus](/guide/blockchain/consensus) (todo: that page doesn't mention "round start" term ).
````
```suggestion
    The start of the round happens when the leader is elected during peer rotation. Generally it happens after the previous block
    is committed. See [Consensus](/guide/blockchain/consensus) (todo: that page doesn't mention "round start" term ).
```

outoftardis commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,67 @@
+
+TODO: Explain Torii module
+
+This module contains configuration of [Torii](/guide/glossary#torii) - the API gateway of Iroha. Refer to the [Torii Endpoints reference](/api/torii-endpoints) for information about exact endpoints.  
```
```suggestion
This module contains configuration of [Torii](/guide/glossary#torii), the API gateway of Iroha. Refer to the [Torii Endpoints reference](/api/torii-endpoints) for detailed information about endpoints.  
```

outoftardis commented on src/api/config/torii-params.md:
```diff
@@ -0,0 +1,67 @@
+- **Type:** String, [Socket-Address](glossary#type-socket-address)
+- **Required**
+
+Address on which Torii Endpoints will be accessible.
```
```suggestion
Address at which Torii Endpoints will be accessible.
```

outoftardis commented on src/api/config/queue-params.md:
```diff
@@ -0,0 +1,67 @@
+# Queue Parameters
+
+TODO Explain Queue module. That is holds or incoming transactions. From time to time, Sumeragi fetches them in order to
```
```suggestion
TODO Explain Queue module. The queue holds incoming transactions. From time to time, Sumeragi fetches them in order to
```

outoftardis commented on src/api/config/queue-params.md:
```diff
@@ -0,0 +1,67 @@
+- **Type:** String or Number, [Duration](glossary#type-duration)
+- **Default:** 1 second
+
+The threshold to determine if a transaction has been tampered to have a future timestamp.
```
```suggestion
The threshold to determine if a transaction has been tampered with to have a future timestamp.
```

outoftardis commented on src/api/config/logger-params.md:
```suggestion
Use compact logging format.
```

outoftardis commented on src/reference/config/glossary.md:
```diff
@@ -0,0 +1,159 @@
+
+## Type: Bytes Amount
+
+Bytes amount is specified as a human-readable string:
```
```suggestion
The amount of bytes is specified as a human-readable string:
```

outoftardis commented on src/reference/config/glossary.md:
````diff
@@ -0,0 +1,159 @@
+value4 = "1GB 5MB"
+```
+
+Iroha can parse sizes in bytes, kilobytes (`K`), megabytes (`M`), gigabytes (`G`), terabytes (`T`),
````
```suggestion
Iroha can parse sizes in bytes (`B`), kilobytes (`K`), megabytes (`M`), gigabytes (`G`), terabytes (`T`),
```

outoftardis commented on src/reference/config/glossary.md:
```diff
@@ -0,0 +1,159 @@
+
+For more information on socket addresses and networking, refer to the [IANA Service Name and Transport Protocol Port Number Registry](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml) or the [RFC 3986](https://tools.ietf.org/html/rfc3986) specification.
+
+## Paths Resolution
```
```suggestion
## Path Resolution
```

that affects links

0x009922 replied:
Actually, we can use [custom anchors](https://vitepress.dev/guide/markdown#custom-anchors):

```md
# Using custom anchors {#my-anchor}
```

0x009922 commented:
@BAStos525, you can preview the migration guide (#481) on the deployment (Reference > Config > Migration): [link](https://iroha-2-docs-m3jkuf9vk-irohas-projects.vercel.app/reference/config/migration.html)

Erigara commented on src/reference/config/network-params.md:
````diff
@@ -0,0 +1,86 @@
+[network]
+address = "localhost:1337"
+```
+
````
I guess new param should go somewhere here: 

```suggestion
## `network.idle_timeout` {#param-idle-timeout}

- **Type:** String or Number, [Duration](glossary#type-duration)
- **Default:** 60 seconds

Duration of time after which connection with peer is terminated if peer is idle.

```toml
[network]
idle_timeout = "60s"
```
```

0x009922 commented:
Updated:

- Reflected changes from https://github.com/hyperledger-iroha/iroha/pull/5384
- Use same tables for typical props of each parameter
- Updated some outdated info

To do:

- Document somewhere on-chain configuration (maybe in a separate PR)
- Do we need some high-level section about peer config?
- There are some undocumented TODOs

@a-zorina need your help.
